src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,386 @@
+        AddPackage(packageList, "dotnet-runtime-", distroType);
+        AddPackage(packageList, "dotnet-targeting-pack-", distroType);
+        AddPackage(packageList, "aspnetcore-runtime-", distroType);
+        AddPackage(packageList, "aspnetcore-targeting-pack-", distroType);
+        AddPackage(packageList, "dotnet-apphost-pack-", distroType);
+        if (Config.Architecture == "x64")
+        {
+            // netstandard package exists for x64 only
+            AddPackage(packageList, "netstandard-targeting-pack-", distroType);
+        }
+        AddPackage(packageList, "dotnet-sdk-", distroType);
+
+        string dockerfile = GenerateDockerfile(packageList, baseImage, distroType);
+
+        string tag = $"test-{Path.GetRandomFileName()}";
+        string output = "";
+
+        try
+        {
+            // Build docker image and run the tests
+            _dockerHelper.Build(tag, dockerfile: dockerfile, contextDir: _contextDir);
+            output = _dockerHelper.Run(tag, tag);
+
+            int testResultsSummaryIndex = output.IndexOf("Tests run: ");
+            if (testResultsSummaryIndex >= 0)
+            {
+                string testResultsSummary = output[testResultsSummaryIndex..];
+                Assert.False(AnyTestFailures(testResultsSummary), testResultsSummary);
+            }
+            else
+            {
+                Assert.Fail("Test summary not found");
+            }
+        }
+        catch (Exception e)
+        {
+            if (string.IsNullOrEmpty(output))
+            {
+                output = e.Message;
+            }
+            Assert.Fail($"Build failed: {output}");
+        }
+        finally
+        {
+            _dockerHelper.DeleteImage(tag);
+        }
+    }
+
+    private string GenerateDockerfile(List<string> rpmPackageList, string baseImage, DistroType distroType)
+    {
+        StringBuilder sb = new();
+        sb.AppendLine("FROM " + baseImage);
+        sb.AppendLine("");
+        sb.AppendLine("# Copy NuGet.config");
+        sb.AppendLine($"COPY NuGet.config .");
+
+        sb.AppendLine("");
+        sb.AppendLine("# Copy scenario-tests content");
+        sb.AppendLine($"COPY scenario-tests scenario-tests");
+
+        sb.AppendLine("");
+        sb.AppendLine("# Copy nuget packages");
+        sb.AppendLine($"COPY packages packages");
+
+        sb.AppendLine("");
+        sb.AppendLine("# Copy RPM packages");
+        foreach (string package in rpmPackageList)
+        {
+            sb.AppendLine($"COPY {package} {package}");
+        }
+        sb.AppendLine("");
+        sb.AppendLine("# Install RPM packages and Microsoft.DotNet.ScenarioTests.SdkTemplateTests tool");
+        sb.Append("RUN");
+
+        // TODO: remove --force-all when aspnet package versioning issue have been resolved - https://github.com/dotnet/source-build/issues/4895
+        string packageInstallationCommand = distroType == DistroType.Deb ? "dpkg -i --force-all" : "rpm -i";
```

Comment:
Could we construct these tests to point to a fake local package feed and install the packages via apt/tdnf/zypper instead of dpkg/rpm directly?

Doing so would let us better test actual end user scenarios

Reply:
That's something we could consider in the future. It's not as trivial to now add that to current tests, at least not as easy as adding a local NuGet store. It would be easy to reuse most of the code for those tests.

For now we're automating current manual testing of RPMs an DEBs, in the same way they are executed.

Utilizing correct installation order also verifies the expected package dependencies.